index.js

```diff
@@ -164,37 +164,38 @@ exports.float64 = {
 exports.buffer = {
   preencode (state, b) {
     if (b) {
-      uint.preencode(state, b.length)
-      state.end += b.length
+      uint.preencode(state, b.byteLength)
+      state.end += b.byteLength
     } else {
       state.end++
     }
   },
   encode (state, b) {
     if (b) {
-      uint.encode(state, b.length)
+      uint.encode(state, b.byteLength)
       state.buffer.set(b, state.start)
-      state.start += b.length
+      state.start += b.byteLength
```

Comment:
I'm actually not entirely sure about this; let me just check how it behaves with non-8-bit typed arrays.

Reply:
It doesn't behave, but that fact was previously masked somewhat. Before, using `b.length`:

```js
cenc.encode(cenc.buffer, Uint32Array.from([1025, 1026, 1027]))
// <Buffer 03 01 02 03>
```

After, using `b.byteLength`:

```js
cenc.encode(cenc.buffer, Uint32Array.from([1025, 1026, 1027]))
// <Buffer 0c 01 02 03 [9 random bytes]>
```

Neither case seems desired, so I'm personally inclined to go with `b.byteLength` and leave the behaviour for anything but `Buffer` and `Uint8Array` undefined. Thoughts?

Reply:
Ah, this actually meant to use .length as it’s an areay of 4 byte entries :)

Reply:
Sorry no confused myself